src/bebb/main.mo

```diff
@@ -631,6 +650,44 @@ actor {
     };
   };
 
+  /**
+   * Helper functions to match filters against all Entities
+  */
+  private func matchEntities(caller : Principal, filterCriteria : [Entity.EntityFilterCriterion]) : ?[Entity.Entity] {
+    if (filterCriteria.size() < 1) {
+      return null;
+    };
+    var foundMatches : Iter.Iter<Entity.Entity> = entitiesStorage.vals();
```

Comment:
Rather than overwriting this variable. It might be more clear to save these values in a all_entities variable, then make a new variable for storing the found matches. It is a little confusing at first since technichally the value doesn't start out as foundMatches but in reality all matches

Reply:
true, I changed them naming and added some comments to make it more clear what's going on here